mozilla_schema_generator/common_ping.py

```diff
@@ -49,33 +49,39 @@ def with_description(dtype: dict, comment: str) -> dict:
 
         active_addons = prepend_properties(("environment", "addons", "activeAddons")) \
             + ("additionalProperties", "properties")
-
-        schema.set_schema_elem(
-            prepend_properties(("environment", "settings", "userPrefs")),
-            with_description(
-                string_map,
-                "User preferences - limited to an allowlist defined in `toolkit/components/telemetry/app/TelemetryEnvironment.jsm`",  # NOQA
-            ),
-        )
-        schema.set_schema_elem(
-                prepend_properties(("environment", "system", "os", "version")), string)
-        schema.set_schema_elem(
-                prepend_properties(("environment", "system", "os", "hasSuperfetch")), boolean)
-        schema.set_schema_elem(
-                prepend_properties(("environment", "system", "os", "hasPrefetch")), boolean)
-        schema.set_schema_elem(
-                prepend_properties(("environment", "addons", "theme", "foreignInstall")), integer)
-        schema.set_schema_elem(active_addons + ("foreignInstall",), integer)
-        schema.set_schema_elem(active_addons + ("version",), string)
-        schema.set_schema_elem(active_addons + ("userDisabled",), integer)
-        schema.set_schema_elem(
-            prepend_properties(("environment", "addons", "activeGMPlugins"))
-            + ("additionalProperties", "properties", "applyBackgroundUpdates"),
-            with_description(
-                integer,
-                "Cast into an integer via mozilla-schema-generator. See bug 1611027.",
-            ),
-        )
+        if property_exists(schema.schema, active_addons):
```

Comment:
The `set_schema_elem` calls here are all the same, but we wrap groups of calls in logic that checks for the existence of the struct. This is necessary to support account-ecosystem, which has a limited environment compared to most common pings.